homeassistant/components/switch/rpi_rf.py

```diff
@@ -0,0 +1,94 @@
+"""
+Allows to configure a switch using a 433MHz module via GPIO on a Raspberry Pi.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.rpi_rf/
+"""
+
+import logging
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.const import CONF_VALUE_TEMPLATE
+
+REQUIREMENTS = ['RPi.GPIO==0.6.2', 'rpi-rf==0.9.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Find and return switches controlled by a generic RF device via GPIO."""
+    import rpi_rf
+
+    gpio = config.get('gpio', 17)
+    rfdevice = rpi_rf.RFDevice(gpio)
+
+    switches = config.get('switches', {})
+    devices = []
+    for dev_name, properties in switches.items():
+        devices.append(
+            RPiRFSwitch(
+                hass,
+                properties.get('name', dev_name),
+                rfdevice,
+                properties.get('protocol', None),
+                properties.get('pulselength', None),
+                properties.get('code_on', 0),
```

Comment:
You're right.
My initial thought was to have the user only specify only one of the two entries for one-way devices but after a second thought that would make no sense for actual two-way switches.
Also, a user may specify 0 manually if explicitly wanted.

Adressed with #6866edab46a44eabc922e2baa40aea6b629773c7